server/src/main/java/org/candlepin/resource/HypervisorResource.java

```diff
@@ -108,7 +110,7 @@ public HypervisorResource(ConsumerResource consumerResource,
     @Deprecated
     @Transactional
     public HypervisorCheckInResult hypervisorUpdate(
-        Map<String, List<GuestId>> hostGuestMap, @Context Principal principal,
+        Map<String, ArrayList<GuestId>> hostGuestMap, @Context Principal principal,
```

Comment:
I've changed to ArrayList as the List class does not allow removal of items in the list using an iterator (as done below).